kuma/search/filters.py

```diff
@@ -96,10 +96,10 @@ class SearchQueryBackend(BaseFilterBackend):
     """
     search_operations = [
         # (<query type>, <field>, <boost factor>)
-        ('match', 'title', 6.0),
+        ('match', 'title', 7.2),
         ('match', 'summary', 2.0),
         ('match', 'content', 1.0),
-        ('match_phrase', 'title', 10.0),
+        ('match_phrase', 'title', 12.0),
         ('match_phrase', 'content', 8.0),
     ]
 
```

Comment:
In the query, you should explecitly mention `minimum_should_match=1`, otherwise, though the query does not match any, it will give other results.

Reply:
``minimum_should_match=1`` is unneeded. From [the docs](https://www.elastic.co/guide/en/elasticsearch/reference/5.6/query-dsl-bool-query.html) for ``should`` in a ``bool`` query:

> If the bool query is a filter context or **has neither must or filter** (*emphasis mine*) then at least one of the should queries must match a document for it to match the bool query.

Reply:
@jwhitlock I know about it but Its a filter backend. There can be multiple backend from a single view so other filter backend may add must and filter query.

Reply:
That's an interesting theory, but I don't believe it.

Reply:
I've read the source code for the ``SearchFilterBackend``, I don't believe it is possible to get a ``must`` or ``filter`` in the ``bool`` block.

Reply:
@jwhitlock If you combine 2 bool query, elasticsearch dsl makes a single bool query. You can check it in the view with `to_dict()` method. You will see all the `must`, `filter` and `should` is merged into a single bool query.

Reply:
I have tested with a sample query `"What"`, and elasticsearch dsl made following query. If you see, the filter, must and should all are included in same `bool` block
```javascript
{
  "query": {
    "boosting": {
      "positive": {
        "bool": {
          "filter": [
            {
              "terms": {
                "locale": [
                  "bn-BD",
                  "en-US"
                ]
              }
            }
          ],
          "must": [
            {
              "function_score": {
                "query": {
                  "bool": {
                    "should": [
                      {
                        "match": {
                          "title": {
                            "query": "what",
                            "boost": 6.0
                          }
                        }
                      },
                      {
                        "match": {
                          "summary": {
                            "query": "what",
                            "boost": 2.0
                          }
                        }
                      },
                      {
                        "match": {
                          "content": {
                            "query": "what",
                            "boost": 1.0
                          }
                        }
                      },
                      {
                        "match_phrase": {
                          "title": {
                            "query": "what",
                            "boost": 10.0
                          }
                        }
                      },
                      {
                        "match_phrase": {
                          "content": {
                            "query": "what",
                            "boost": 8.0
                          }
                        }
                      }
                    ]
                  }
                },
                "functions": [
                  {
                    "field_value_factor": {
                      "field": "boost"
                    }
                  }
                ]
              }
            }
          ]
        }
      },
      "negative": {
        "bool": {
          "must_not": [
            {
              "term": {
                "locale": "bn-BD"
              }
            }
          ]
        }
      },
      "negative_boost": 0.5
    }
  },
  "explain": true,
  "highlight": {
    "fields": {
      "content": {
        
      },
      "summary": {
        
      }
    },
    "post_tags": [
      "</mark>"
    ],
    "order": "score",
    "pre_tags": [
      "<mark>"
    ]
  }
}
```

Reply:
I see three ``bool`` blocks, the one we are talking about just has a ``should`` clause.